### PR TITLE
ERC20Cogateway subgraph setup

### LINKED ISSUE #276: Setup schema.graphql, mapping(handler), schema, subgraph.yml file for ERC20Cogateway contract
- Define the entities in schema.graphql

```
DeclaredWithdrawIntent
      - id
      - token address
      - amount
      - nonce
      - beneficiary
      - fee gas price
      - fee gas limit
      - withdrawer
      - block number
      - block hash
      - contract address
      - updated time stamp (uts)
ConfirmedDepositIntent
      - id
      - message hash
      - block number
      - block hash
      - contract address
      - updated time stamp (uts)
ProvenGateway
      - id
      - proven block number
      - storage root
      - gateway address 
      - block number
      - block hash
      - contract address
      - updated time stamp (uts)
CreatedUtilityToken
      - id
      - value token
      - utility token
      - block number
      - block hash
      - contract address
      - updated time stamp (uts)
```
- Define handlers 
  - ConfirmedDepositIntent
  - DeclaredWithdrawIntent
  - CreatedUtilityToken, 
  - ProvenGateway
- Add ABIs for
  - ERC20Cogateway contract
- Events in subgraph.yaml
  - DeclaredWithdrawIntent
  - ConfirmedDepositIntent
  - GatewayProven event
  - UtilityTokenCreated event

### PR DESCRIPTION
**Pull request by gulshanvasnani**
fixes #276

### CODE REVIEW

File: subgraph/auxiliary/src/ERC20CogatewayMapping.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  entity.blockNumber = event.block.number;
+  entity.blockHash = event.block.hash;
+  entity.contractAddress = event.address;
+  entity.uts = event.block.timestamp;
+  entity.save();
+}
+
+export function handleGatewayProven(event: GatewayProven): void {
+  let entity = new ProvenGateway(
+    // eslint-disable-next-line prefer-template
```
Comment (abhayks1) [inline subgraph/auxiliary/src/ERC20CogatewayMapping.ts]: this is duplicate. We have this at top of file already. Please check and remove it.

File: subgraph/auxiliary/package-lock.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "mosaic-subgraph",
```
Comment (abhayks1) [inline subgraph/auxiliary/package-lock.json]: I don't see package.json updating with correct name

File: subgraph/auxiliary/subgraph.yaml [hunk 3]
```diff
@@ -21,3 +21,32 @@ dataSources:
         - event: StateRootAvailable(uint256,bytes32)
           handler: handleStateRootAvailable
       file: ./src/AnchorMapping.ts
+
+  - kind: ethereum/contract
+    name: ERC20Cogateway
+    network: dev
+    source:
+      address: "0xdbbb5c822ccfac3ea49e32215565c6aafb257a30"
```
Comment (abhayks1) [inline subgraph/auxiliary/subgraph.yaml:29]: Update in ticket that contract address needs to be updated after deployment and tag @deepesh-kn

**Comment by abhayks1** [comment]
Please run below commands after syncing the graph node
```
npm run create-local
```

**Review by abhayks1 — CHANGES_REQUESTED** [review]
There are few inline comments

**Comment by gulshanvasnani** [comment]
> Please run below commands after syncing the graph node
> 
> ```
> npm run create-local
> ```

Thank you Abhay.
I have run the above command and subgraph gets created with following output on console
```

> mosaic-subgraph@ create-local /Users/gulshanvasnani/Documents/forks/facilitator/subgraph/auxiliary
> graph create --node http://172.16.0.191:9425 mosaic/aux-erc20-gateways

Created subgraph: mosaic/aux-erc20-gateways
```

File: subgraph/auxiliary/subgraph.yaml (hunk 3 above)
Comment (gulshanvasnani) [reply subgraph/auxiliary/subgraph.yaml:29]: Commented on the ticket.

File: subgraph/auxiliary/package-lock.json (hunk 2 above)
Comment (gulshanvasnani) [reply subgraph/auxiliary/package-lock.json]: Done.

File: subgraph/auxiliary/src/ERC20CogatewayMapping.ts (hunk 1 above)
Comment (gulshanvasnani) [reply subgraph/auxiliary/src/ERC20CogatewayMapping.ts]: Done.

**Comment by abhayks1** [comment]
Please take latest develop changes once and test the subgraphs by deploying on graph nodes.

**Comment by gulshanvasnani** [comment]
> Please take latest develop changes once and test the subgraphs by deploying on graph nodes.

Done

**Review by abhayks1 — APPROVED** [review]
LGTM 👍

**Pull request merged**